File: 800/CF2248A_You_Delete_I_Delete.py
t = int(input())

for _ in range(t):
    s = input()

    # Alice removes the first 0
    first_zero = s.index('0')
    after_alice = s[:first_zero] + s[first_zero + 1:]

    # Bob chooses which 1 to remove
    remove_one = -1

    for i in range(len(after_alice)):
        if after_alice[i] == '1':
            if '0' in after_alice[i + 1:]:
                remove_one = i
                break

    # If every remaining 1 is at the end,
    # remove the last 1.
    if remove_one == -1:
        remove_one = after_alice.rfind('1')

    answer = after_alice[:remove_one] + after_alice[remove_one + 1:]

    print(answer)

File: 800/CF2248A_You_Delete_I_Delete.java

import java.util.Scanner;

public class CF2248A_You_Delete_I_Delete {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int testCases = sc.nextInt();

        while (testCases-- > 0) {

            String s = sc.next();

            // Alice removes the first 0
            int firstZero = s.indexOf('0');

            StringBuilder afterAlice = new StringBuilder();

            for (int i = 0; i < s.length(); i++) {
                if (i != firstZero) {
                    afterAlice.append(s.charAt(i));
                }
            }

            // Bob decides which 1 to remove
            int removeOne = -1;

            for (int i = 0; i < afterAlice.length(); i++) {

                if (afterAlice.charAt(i) == '1') {

                    boolean zeroLater = false;

                    for (int j = i + 1; j < afterAlice.length(); j++) {
                        if (afterAlice.charAt(j) == '0') {
                            zeroLater = true;
                            break;
                        }
                    }

                    if (zeroLater) {
                        removeOne = i;
                        break;
                    }
                }
            }

            // If no such 1 exists, remove the last 1
            if (removeOne == -1) {
                for (int i = afterAlice.length() - 1; i >= 0; i--) {
                    if (afterAlice.charAt(i) == '1') {
                        removeOne = i;
                        break;
                    }
                }
            }

            StringBuilder answer = new StringBuilder();

            for (int i = 0; i < afterAlice.length(); i++) {
                if (i != removeOne) {
                    answer.append(afterAlice.charAt(i));
                }
            }

            System.out.println(answer);
        }

        sc.close();
    }
}